pages/admin/index.tsx

```diff
@@ -16,61 +16,53 @@ import {
   InputGroup,
   Input,
   InputLeftElement,
-  useControllableState,
   Wrap,
 } from '@chakra-ui/react'; // Chakra UI
 import { ChevronDownIcon, SearchIcon } from '@chakra-ui/icons'; // Chakra UI Icons
 import Layout from '@components/internal/Layout'; // Layout component
-import { Role } from '@lib/types'; // Role enum
 import { authorize } from '@tools/authorization'; // Page authorization
 import Table from '@components/internal/Table'; // Table component
 import Pagination from '@components/internal/Pagination'; // Pagination component
 import RequestStatusBadge from '@components/internal/RequestStatusBadge'; //Status badge component
+import { useQuery } from '@apollo/client'; //Apollo client
+import { useEffect, useState } from 'react'; // React
+import {
+  GetApplicationsRequest,
+  GetApplicationsResponse,
+  GET_APPLICATIONS_QUERY,
+} from '@tools/pages/admin'; //Applications queries
+import { SortOptions, SortOrder } from '@tools/types'; //Sorting types
+import { ApplicationStatus, PermitType, Role } from '@lib/graphql/types'; //GraphQL types
+import useDebounce from '@tools/hooks/useDebounce'; // Debounce hook
+import { Column } from 'react-table';
 
-type StatusProps = {
-  readonly value:
-    | 'COMPLETED'
-    | 'INPROGRESS'
-    | 'PENDING'
-    | 'REJECTED'
-    | 'EXPIRING'
-    | 'EXPIRED'
-    | 'ACTIVE';
+// Map uppercase enum strings to lowercase
+const permitTypeString: Record<PermitType, string> = {
+  [PermitType.Permanent]: 'Permanent',
+  [PermitType.Temporary]: 'Temporary',
 };
 
-function renderStatusBadge({ value }: StatusProps) {
-  return (
-    <Wrap>
-      <RequestStatusBadge variant={value}></RequestStatusBadge>
-    </Wrap>
-  );
-}
-
-type NameProps = {
-  readonly value: {
-    name: string;
-    id: string;
-  };
+type PermitTypeProps = {
+  value: PermitType;
 };
 
-function renderName({ value }: NameProps) {
-  return (
-    <div>
-      <Text>{value.name}</Text>
-      <Text textStyle="caption" textColor="secondary">
-        ID: {value.id}
-      </Text>
-    </div>
-  );
-}
-
 // Placeholder columns
-// TODO: accessors should be the accessors for these fields in the DB
-const COLUMNS = [
+const COLUMNS: Column<any>[] = [
   {
     Header: 'Name',
     accessor: 'name',
-    Cell: renderName,
+    Cell: ({ value }) => {
+      return (
+        <div>
+          <Text>{value.firstName + ' ' + value.lastName}</Text>
```

Comment:
```suggestion
          <Text>{`${value.firstName} ${value.lastName}`}</Text>
```
JS string interpolation